File: src/controller/TagController.js
var url = require("url"),
		util = require("util");

module.exports.controller = function(app) {
	var tagM = require("./../model/TagModel")(app);
	app.get('/api/tag/search', function(req, res) {
		var queryData = url.parse(req.url, true).query;
		tagM.searchTagsByName(queryData.q, function(err, tags){
			if(err){
				console.error('m=searchTagsByName, err=%s', err);
				app.em._500({
					message: util.format("Erro ao buscar tags por: %s", queryData.q),
					stacktrace: err,
					res: res
				});
			}else{
				res.send(tags);
			}
		});
	});
	app.put('/api/tag', function(req, res) {
		tagM.insertTag(req.body, function(err, tags){
			res.send();
		});
	});
	app.get('/api/tag', function(req, res) {
		tagM.getTags(function(err, rows){
			if(err){
				app.em._500({
					message: "Opa, encontramos um problema ao carregar os bookmarks",
					stacktrace: err.stack,
					res: res
				});
			}else{
				res.send(rows);
			}
		});
	});

};

File: src/main/java/com/mageddo/bookmarks/apiserver/TagController.java
package com.mageddo.bookmarks.apiserver;

import com.mageddo.bookmarks.service.TagService;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.QueryValue;

import static io.micronaut.core.util.CollectionUtils.mapOf;
import static io.micronaut.http.HttpResponse.ok;
import static io.micronaut.http.HttpResponse.serverError;

@Controller
public class TagController {

  private final Logger logger = LoggerFactory.getLogger(getClass());
  private final TagService tagService;

  public TagController(TagService tagService) {
    this.tagService = tagService;
  }

  @Get("/api/tag")
  public HttpResponse _1() {
    try {
      return ok(tagService.getTags());
    } catch (Exception e) {
      logger.error("status=failed-load-tags, msg={}", e.getMessage(), e);
      return serverError(mapOf("message", "Opa, encontramos um problema ao carregar os bookmarks"));
    }
  }

  @Get("/api/tag/search")
  HttpResponse _2(@QueryValue("q") String query) {
    try {
      return ok(tagService.findTags(query));
    } catch (Exception e) {
      logger.error("status=cant-filter-tags, msg={}", e.getMessage(), e);
      return serverError(mapOf("mesage", "Erro ao buscar tags por: " + query)).header("caught", "1");
    }
  }
}
